### PR TITLE
add ablility to like an album/song

### PR DESCRIPTION
**Pull request by alls77**

### CODE REVIEW

File: backend/Perflow/Common/Helpers/LikedSong.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+        public static implicit operator (Song s, bool)(LikedSong value)
+        {
+            return (value.Song, value.IsLiked);
+        }
+
+        public static implicit operator LikedSong((Song s, bool) value)
+        {
+            return new LikedSong(value.s, value.Item2);
+        }
```
Comment (proskurnicha) [inline backend/Perflow/Common/Helpers/LikedSong.cs]: Why do we need implicit operator?
Could we get rid of it and use a mapper or constructor instead?

File: backend/Perflow/Common/DTO/Albums/AlbumFullDTO.cs [hunk 2]
```diff
@@ -9,16 +9,18 @@ namespace Perflow.Common.DTO.Albums
...
     {
         public int Id { get; set; }
-        public int ReleaseYear { get; set; }
+        public int? ReleaseYear { get; set; }
         public string Description { get; set; }
         public AuthorType AuthorType { get; set; }
         public bool IsPublished { get; set; }
         public AlbumRegion Region { get; set; }
         public bool IsSingle { get; set; }
+
```
Comment (proskurnicha) [inline backend/Perflow/Common/DTO/Albums/AlbumFullDTO.cs]: Delete empty line

File: backend/Perflow/Services/Implementations/AlbumsService.cs [hunk 3]
```diff
@@ -32,21 +35,39 @@ public async Task<Album> GetEntityAsync(int id)
...
                                             .AsNoTracking()
+                                            .Select(a => new AlbumFullDTO
+                                            {
+                                                Id = a.Id,
+                                                Name = a.Name,
+                                                ReleaseYear = a.ReleaseYear,
+                                                IconURL = a.IconURL,
+                                                Songs = a.Songs
+                                                .Select(s =>
+                                                mapper.Map<LikedSong, SongReadDTO>(new LikedSong(s, s.Reactions.Any(r => r.UserId == userId)))
```
Comment (proskurnicha) [inline backend/Perflow/Services/Implementations/AlbumsService.cs]: Add extra tab here

File: backend/Perflow/Services/Implementations/SongsService.cs [hunk 4]
```diff
@@ -36,10 +36,12 @@ public async Task<IEnumerable<SongReadDTO>> GetLikedSongsAsync(int userId)
                     .ThenInclude(song => song.Group)
                 .Include(songReaction => songReaction.Song)
                     .ThenInclude(song => song.Album)
-                .Select(songReaction => songReaction.Song)
+                .Select(songReaction =>
+                  mapper.Map<LikedSong, SongReadDTO>(new LikedSong(songReaction.Song, true))
```
Comment (proskurnicha) [inline backend/Perflow/Services/Implementations/SongsService.cs]: Add tab here, as we have here:
image

File: frontend/src/app/modules/album/components/album-details/album-details.component.ts [hunk 5]
```diff
@@ -15,18 +18,21 @@ import { ClipboardService } from 'ngx-clipboard';
...
 
   @ViewChild('albums') albumsElement: ElementRef;
   album: AlbumFull = {} as AlbumFull;
   anotherAlbums: Album[] = [];
 
   constructor(private _clipboardApi: ClipboardService, private _route: ActivatedRoute, private _service: AlbumService,
-    private _router: Router, private _location: PlatformLocation) {
+    private _reactionService: ReactionService, private _router: Router, private _location: PlatformLocation,
+    private _authService: AuthService) {
     this._router.routeReuseStrategy.shouldReuseRoute = () => false;
```
Comment (proskurnicha) [inline frontend/src/app/modules/album/components/album-details/album-details.component.ts:36]: Use next formatting 
check https://github.com/BinaryStudioAcademy/quality-criteria
image

**Review by proskurnicha — CHANGES_REQUESTED** [review]

File: backend/Perflow/Common/Helpers/LikedSong.cs (hunk 1 above)
Comment (alls77) [reply backend/Perflow/Common/Helpers/LikedSong.cs]: sorry, i forgot to delete it

**Pull request merged**